Added RandomSizedCropTorchVision

vfdev-5 opened the pull request:
WIP PR to fix #259


cc @ternaus

albu commented on albumentations/augmentations/transforms.py:
```diff
@@ -717,7 +718,112 @@ def apply_to_keypoint(self, keypoint, crop_height=0, crop_width=0, h_start=0, w_
+        return ['image']
+
+
+def RandomSizedCrop(*args, mode='old', **kwargs):
```
looks like we can pass height and width for both transforms so no need to use args/kwargs for them. for other arguments it seems like we can add here all params (with same defaults) from new version and just use args, kwargs for old.

vfdev-5 replied:
I see what what you mean, sounds good. What about `mode='old'` and "add here all params (with same defaults) from new version", this would be a bit strange ?

albu replied:
yeah it's a bit strange but doesn't break anything (but changing to mode='torchvision' breaks). at the same time it tries to push users to new api (gently).

albu replied:
oh, I see what happens, agree with you - it's strange and weird :(

albu replied:
so new suggestion. if user passes `min_max_height` - run old version. otherwise run new. how it sounds?

vfdev-5 replied:
I'm not sure how to determine whether user passes `min_max_height`. For example, in your tests:
https://github.com/albu/albumentations/blob/448761df9a008384cf914343f25e3cfb7c4d7551/tests/test_bbox.py#L189

In this case we need to check what type of input is passed. So decision based on type of input seems not very safe ?

vfdev-5 replied:
> yeah it's a bit strange but doesn't break anything (but changing to mode='torchvision' breaks). at the same time it tries to push users to new api (gently).

Yes, you are right, we need to keep `mode='old'` to avoid BC breaks.

vfdev-5 commented:
I refactored transforms to avoid some deprecation warnings with `RandomSizedCropTorchVision` and updated input args of `RandomSizedCrop`.

@albu Are you ok with that `RandomSizedCrop` is a **method** and not a class ?

albu commented:
> I refactored transforms to avoid some deprecation warnings with `RandomSizedCropTorchVision` and updated input args of `RandomSizedCrop`.
> 
> @albu Are you ok with that `RandomSizedCrop` is a **method** and not a class ?

it could break documentation and probably something else.

I wonder if just adding new crop with different name could be better and safer solution..

vfdev-5 commented:
> I wonder if just adding new crop with different name could be better and safer solution..

I agree. I let you decide between maintainers what names to choose.

albu commented:
maybe `RandomResizedCrop`?

ternaus commented:
It looks like Torchvision deprecated `RandomSizedCrop` in favor of `RandomResizedCrop`

I would love to see:
1. Right now (0.3) to have `RandomSizedCrop` that we had earlier.
2. `RandomResizedCrop` with the API and functionality of what torchvision has https://pytorch.org/docs/stable/torchvision/transforms.html#torchvision.transforms.RandomResizedCrop
3. Deprecation warning in `RandomSizedCrop`

=> in 0.4 we will match the behavior of Torchvision, keeping users informed of the future breaking changes.

So, to me, looks like adding second transform `RandomResizedCrop` is the best solution.

vfdev-5 commented:
@ternaus review, please :)

ternaus commented on albumentations/augmentations/transforms.py:
```diff
@@ -688,16 +712,12 @@ class RandomSizedCrop(DualTransform):
-    def apply(self, img, crop_height=0, crop_width=0, h_start=0, w_start=0, interpolation=cv2.INTER_LINEAR, **params):
-        crop = F.random_crop(img, crop_height, crop_width, h_start, w_start)
-        return F.resize(crop, self.height, self.width, interpolation)
+        warnings.warn("This class has been deprecated. Please use RandomSizedCropTorchVision", DeprecationWarning)
```
Could you please fix this warning?

vfdev-5 replied:
Done
